lib/jsonapi/authorization/default_pundit_authorizer.rb

```diff
@@ -245,6 +292,23 @@ def authorize_relationship_operation(source_record, relationship_method, *args)
           ::Pundit.authorize(user, source_record, 'update?')
         end
       end
+
+      def relationship_method(data, **options)
+        relationship = data[:relationship]
+        assoc_name   = data[:relation_name]
+        prefix       = options[:prefix]
+
+        case relationship
+        when ->(relationship) { relationship.polymorphic }
+          polymorphic_type = data[:records].class.name.downcase
+          "#{prefix}_#{relationship.class_name.downcase}_#{polymorphic_type}?"
+        when ->(relationship) { relationship.class == JSONAPI::Relationship::ToOne }
```

Comment:
I think these case-statements could be written like so?

```rb
case relationship
when relationship.polymorphic
  # ...
when JSONAPI::Relationship::ToOne
  # ...
else
  # ...
end
```

I haven't seen this sort of lambda usage before myself 😅